### PR TITLE
Fixing to only send one solr query for restricted totals

### LINKED ISSUE #1003: Replace subtraction logic in ResourceStatisticsSource
The `ResourceStatisticsSource` (`app/sources/curation_concerns/resource_statistics_source.rb`) currently gets a count restricted objects by querying for total objects and non-restricted objects and then calculating the difference. If we can get Solr to do our bidding, it would be preferable for Solr to do this calculation for us. We need a count of all Solr documents that either 1) lack the `read_access_group_ssim` key or 2) have the key without the values of `public` or `registered` (if the object is shared with other groups).

### PR DESCRIPTION
**Pull request by cam156**
Fixes #1003 

Use a negative query instead of making three queries to solr.

@projecthydra/sufia-code-reviewers

### CODE REVIEW

File: app/sources/curation_concerns/resource_statistics_source.rb [hunk 1]
```diff
@@ -10,60 +10,47 @@ def initialize(search_builder: ::CurationConcerns::AdminController.new.search_bu
...
-      results_count("#{Hydra.config.permissions.read.group}:registered")
+      results_count(registered_read_group)
     end
 
     def restricted_concerns_count
-      # TODO: Replace this with a query that that returns all documents that
-      #       either lack the `read_access_group_ssim` key, or have the key
-      #       without the values of `public` or `registered`
-      repository.search(solr_arguments)["response"]["numFound"] - (authenticated_concerns_count + open_concerns_count)
+      results_count([not_regisitered_read_group,not_public_read_group])
```
Comment (mjgiarlo) [inline app/sources/curation_concerns/resource_statistics_source.rb]: `s/regisitered/registered/`

File: app/sources/curation_concerns/resource_statistics_source.rb [hunk 2]
```diff
@@ -10,60 +10,47 @@ def initialize(search_builder: ::CurationConcerns::AdminController.new.search_bu
...
     end
 
     def expired_embargo_now_authenticated_concerns_count
-      results_count(["#{Hydra.config.permissions.read.group}:registered", "embargo_history_ssim:*"])
+      results_count([registered_read_group, embargo_history_query])
```
Comment (mjgiarlo) [inline app/sources/curation_concerns/resource_statistics_source.rb]: These are all queries -- why have some methods that end in `_query` and some that don't? Could you `s/_query//g` ?

File: app/sources/curation_concerns/resource_statistics_source.rb [hunk 3]
```diff
@@ -10,60 +10,47 @@ def initialize(search_builder: ::CurationConcerns::AdminController.new.search_bu
...
     end
 
     def active_embargo_now_restricted_concerns_count
-      # TODO: Replace the subtraction with another `#where` query that returns
-      #       all actively embargoed documents that either lack the
-      #       `read_access_group_ssim` key, or have the key without the values
-      #       of `public` or `registered`
-      all_expired_embargos = results_count("embargo_release_date_dtsi:[NOW TO *]")
-      all_expired_embargos - active_embargo_now_authenticated_concerns_count
+      results_count([not_regisitered_read_group, not_public_read_group, active_embargo])
```
Comment (mjgiarlo) [inline app/sources/curation_concerns/resource_statistics_source.rb]: `s/regisitered/registered/`

File: app/sources/curation_concerns/resource_statistics_source.rb [hunk 4]
```diff
@@ -10,60 +10,47 @@ def initialize(search_builder: ::CurationConcerns::AdminController.new.search_bu
...
     end
 
     def expired_lease_now_restricted_concerns_count
-      # TODO: Replace the subtraction with another `#where` query that returns
-      #       all expired lease documents that either lack the
-      #       `read_access_group_ssim` key, or have the key without the values
-      #       of `public` or `registered`
-      all_leased_documents = results_count("lease_history_ssim:*")
-      all_leased_documents - expired_lease_now_authenticated_concerns_count
+      results_count([not_regisitered_read_group, not_public_read_group, lease_history_query])
```
Comment (mjgiarlo) [inline app/sources/curation_concerns/resource_statistics_source.rb]: `s/regisitered/registered/`

File: app/sources/curation_concerns/resource_statistics_source.rb [hunk 5]
```diff
@@ -78,5 +65,38 @@ def results_count(query)
...
+
+      def embargo_history_query
+        "embargo_history_ssim:*"
+      end
+
+      def registered_read_group
+        "#{Hydra.config.permissions.read.group}:registered"
+      end
+
+      def not_regisitered_read_group
```
Comment (mjgiarlo) [inline app/sources/curation_concerns/resource_statistics_source.rb]: `s/regisitered/registered/`

File: spec/sources/curation_concerns/resource_statistics_source_spec.rb [hunk 6]
```diff
@@ -49,6 +51,9 @@
     end
 
     describe "#active_embargo_now_authenticated_concerns_count" do
+      before do
+        create :embargoed_work, embargo_date: Date.yesterday.to_s, current_state: current_state, future_state: future_state
```
Comment (mjgiarlo) [inline spec/sources/curation_concerns/resource_statistics_source_spec.rb]: Won't `Date.yesterday.to_s` create an expired embargo?

Comment (cam156) [inline spec/sources/curation_concerns/resource_statistics_source_spec.rb]: Yes.  I wanted to be certain I only got the active one

**Pull request merged**